app/src/main/java/ar/scacchipa/twittercloneapp/repository/CredentialRepository.kt

```diff
@@ -0,0 +1,43 @@
+package ar.scacchipa.twittercloneapp.repository
+
+import android.content.SharedPreferences
+import ar.scacchipa.twittercloneapp.data.Credential
+import ar.scacchipa.twittercloneapp.utils.Constants
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+class CredentialRepository(
+    private val credentialLocalSource: SharedPreferences,
+    private val dispatcher: CoroutineDispatcher = Dispatchers.Default
+): ICredentialRepository {
+
+    override fun recoverCredential(): Credential {
```

Comment:
I think a better approach would be returnning a Credential? (nullable) type instead of returning an empty credential.